spec/rubocop/cop/rspec/empty_hook_spec.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::RSpec::EmptyHook do
+  subject(:cop) { described_class.new(config) }
+
+  let(:config) { RuboCop::Config.new }
+
+  shared_examples 'empty hook' do |hook_type|
+    it 'registers offense with empty block' do
+      expect_offense(<<~RUBY)
+        #{hook_type} {}
+        #{'^' * hook_type.length} Empty hook detected.
+      RUBY
+    end
+
+    it 'registers offense with empty single-line do-end' do
+      expect_offense(<<~RUBY)
+        #{hook_type} do; end
+        #{'^' * hook_type.length} Empty hook detected.
+      RUBY
+    end
+
+    it 'registers offense with empty multi-line do-end' do
+      expect_offense(<<~RUBY)
+        #{hook_type} do
+        #{'^' * hook_type.length} Empty hook detected.
+        end
+      RUBY
+    end
+
+    it 'registers offense with empty single-line nil' do
+      expect_offense(<<~RUBY)
+        #{hook_type} { nil }
+        #{'^' * hook_type.length} Empty hook detected.
+      RUBY
+    end
+
+    it 'registers offense with empty multi-line nil' do
+      expect_offense(<<~RUBY)
+        #{hook_type} do
+        #{'^' * hook_type.length} Empty hook detected.
+          nil
+        end
+      RUBY
+    end
+
+    it 'allows single-line non-empty blocks' do
+      expect_no_offenses(<<~RUBY)
+        #{hook_type} { create_users }
+      RUBY
+    end
+
+    it 'allows multi-line non-empty blocks' do
+      expect_no_offenses(<<~RUBY)
+        #{hook_type} do
+          create_users
+          create_businesses
+        end
+      RUBY
+    end
+  end
+
+  context 'with before hooks' do
+    it_behaves_like 'empty hook', 'before'
+    it_behaves_like 'empty hook', 'before(:each)'
+    it_behaves_like 'empty hook', 'before(:suite)'
+    it_behaves_like 'empty hook', 'before(:all)'
+  end
+
+  context 'with after hooks' do
+    it_behaves_like 'empty hook', 'after'
+    it_behaves_like 'empty hook', 'after(:each)'
+    it_behaves_like 'empty hook', 'after(:suite)'
+    it_behaves_like 'empty hook', 'after(:all)'
+  end
```

Comment:
Even though this suite is comprehensive and concise, it has redundancy and is quite hard to read in regards to using string interpolation.
WDYT of writing it instead as:
```ruby
it 'detects an offense for an empty `before`'
  ...
end

it 'detects the offense for an empty `before` with a scope' do
  ...
end

it 'detects ... `after`' do
it 'detects ... `prepend_after` do
it 'detects `:all` scope' do
it 'detects `:context' scope' do
```
etc?